_ap/17/index.adoc

```diff
@@ -0,0 +1,60 @@
+---
+num: 17
+tags:
+  - "tls"
+  - "certificates"
+  - "dns"
+  - "Letsencrypt"
+title: "Automatic management of certificates per managed instance using Letsencrypt"
+status: "Accepted"
+authors:
+  - "Manyanda Chitimbo"
+---
+
+## Intent
+
+Define a pattern for the automatic management of certificates per managed instance using Letsencrypt
+
+## Motivation
+
+When a managed service is deployed on the data plane cluster its routes have be reachable via TLS.
+It means that a managed service deployed in the data plane has to provide a way to the clients (end users) to verify and validate the TLS connection.
+
+## Applicability
+
+This pattern applies to any service deployed on a dataplane that wants a managed service routes to be reachable via TLS
+
+## Description
+
+A managed service will use https://Letsencrypt.org[Letsencrypt] to managed certificates on the fly i.e automatic and thus removing the need for manual intervention and the toil related to this.
+
+To avoid sharing of certificates between deployed instances, managed service implementors are recommended to generate a wildcard certificate per managed instance.
+
+As an example, if the root domain is `myservice.bf2.dev`, then a wildcard certificate will be generated for each new managed instance under a subdomain `<managed-instance-id>.myservice.bf2.dev`. The `<managed-instance-id>` is the unique id of the managed instance.
+
+Furthermore, all routes for the managed services deployed on the the data plane will be of the form `<unique-route-slug>.<managed-instance-id>.myservice.bf2.dev`.
+If we the Managed Kafka Service as an example, it will expose Kafka routes with the form `<unique-route-slug>.<kafka-instance-id>.kafka.bf2.dev` e.g `boostrapserverhost.123456.kafka.bf2.dev`.
+
```

Comment:
should we qualify as in if there is a possibility of using multiple subdomains as hosts under a given instance id then a wildcard per instance id is recommended. If there is only ever a single subdomain then there is no need for a wildcard. I guess what I am saying is I am not sure we should force the use of a wildcard if a single domain cert would suffice

Reply:
The suggestion make sense. I can reword it in a way you are suggesting.